scripts/fuse/elektra_fuse/elektra_util.py

```diff
@@ -0,0 +1,203 @@
+from pathlib import Path
+import os, re, errno
+
+import kdb
+
+
+#TODO: Add 'default:' and 'proc:'. These namespaces do not work out of the box like "user:", "sysem:", ... but need extra treatment.
+elektra_namespaces = ["user:", "system:", "dir:", "spec:", "cascading:"]
+
+dir_file_special_name = "®elektra.value"
+
+#translates from filesystem (that are below the "pid"-level) paths to elektra paths (e.g. '/user:/dir/@elektra.value' -> 'user:/dir', '/cascading:/key' -> '/key') 
+def os_path_to_elektra_path(os_path):
+    if Path(os_path).name == dir_file_special_name:
+        elektra_path = str(Path(os_path).parent).strip("/")
+    elif os_path.startswith("/cascading:"):
+        elektra_path = os_path[len("/cascading:"):] #could use removeprefix with python 3.9+
+        if elektra_path == "":
+            elektra_path = "/"
+    else:
+        elektra_path = os_path.strip("/") #remove slashes ('/' is reserved for the cascading namespace)
+    
+    if elektra_path.endswith(":"):
+        elektra_path = elektra_path + "/" #special case intruced around elektra v5 (the root of a namespace needs a trailing slash)
+
+    return elektra_path
+    
+
+def size_of_file(os_path):
+    return len(file_contents(os_path))
+
+def is_directory_empty(os_path):
+    dirs, files = ls(os_path)
+    return not bool(dirs) and not bool(files)
+   
+#performs function of the "kdb file" command 
+def get_kdb_file(os_path):
+    with kdb.KDB() as db:
+        elektra_path = os_path_to_elektra_path(os_path)
+        x = kdb.Key(elektra_path)
+        db.get(kdb.KeySet(), x)
+        return x.value 
+
+def update_key_value(os_path: str, new_value: bytes):
+    # kdb.kdb.KDBException, may be thrown
+    # validation => whole key needs to be written at once
+
+    with kdb.KDB() as db:
+        path = os_path_to_elektra_path(os_path)
+
+        ks = kdb.KeySet()
+        db.get(ks, path)
+        key = ks[path]
+
+
+        #try to save new_value as UTF-8 string in case it can be decoded as such
+        #TODO: manage 'binary' meta-key
+        try:
+            new_value_as_string = new_value.decode(encoding="utf-8", errors="strict")
+            key.value = new_value_as_string
+        except UnicodeDecodeError:
+            key.value = new_value
+
+        db.set(ks, path) #using key instead of path here deleted the key
+
+
+#unexpected behavior, this deletes child keys too (why?)
```

Comment:
> this deletes child keys too (why?)

Because you used `ks.cut (key)` which is defined as "remove everything below (and including) `key` from `ks`, put it into a new KeySet and return that".

If you just want to delete a single key by name, you should use `keyDel(ksLookupByName (ks, name, KDB_O_POP))` (not sure how exactly that is exposed in Python).

Reply:
Removed the offending method in 378c276ca07183e62180a72391eaaafa93b12c97. (kdb is used directly as a remedy)

Reply:
Please create issues if there are problems with the Python binding. I created an issue now: #3749

@dev2718 wrote:
> kdb is used directly as a remedy

What do you mean? Is there some fallback where the `kdb` CLI tool is used?